app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java

```diff
@@ -85,6 +89,50 @@ public static boolean isTv(final Context context) {
         return DeviceUtils.isTV;
     }
 
+    public static boolean isDesktopMode(final Context context) {
+        if (ContextCompat.getSystemService(context, UiModeManager.class)
+                .getCurrentModeType() == Configuration.UI_MODE_TYPE_DESK) {
+            return true;
+        }
+        // DeX check for standalone and multi-window mode
+        try {
+            final Configuration config = context.getResources().getConfiguration();
+            final Class<?> configClass = config.getClass();
+            final int semDesktopModeEnabledConst =
+                    configClass.getField("SEM_DESKTOP_MODE_ENABLED").getInt(configClass);
+            final int currentMode =
+                    configClass.getField("semDesktopModeEnabled").getInt(config);
+            if (semDesktopModeEnabledConst == currentMode) {
+                return true;
+            }
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            // empty
+        }
+        @SuppressLint("WrongConstant") final Object desktopModeManager = context
+                .getApplicationContext()
+                .getSystemService("desktopmode");
+        if (desktopModeManager != null) {
+            try {
+                final Method getDesktopModeStateMethod = desktopModeManager.getClass()
+                        .getDeclaredMethod("getDesktopModeState");
+                final Object desktopModeState = getDesktopModeStateMethod
+                        .invoke(desktopModeManager);
+                final Class<?> desktopModeStateClass = desktopModeState.getClass();
+                final Method getEnabledMethod = desktopModeStateClass
+                        .getDeclaredMethod("getEnabled");
+                final int enabled = (int) getEnabledMethod.invoke(desktopModeState);
+                if (enabled == desktopModeStateClass
+                        .getDeclaredField("ENABLED").getInt(desktopModeStateClass)) {
+                    return true;
+                }
+            } catch (NoSuchFieldException | NoSuchMethodException
+                    | IllegalAccessException | InvocationTargetException e) {
+                // Device does not support DeX 3.0
+            }
+        }
+        return false;
```

Comment:
Where did you get these values from?

Reply:
Samsung DeX check code can be found from Samsung's developer guide:
https://developer.samsung.com/samsung-dex/modify-optimizing.html